### ISSUE TITLE
Change the way mouse input is handled

### ISSUE THREAD

**Issue by lunastrans** [catsout/wallpaper-engine-kde-plugin#64]
Hello. I love this plugin. But I have a slight issue with how the mouse input is handled. 
Mouse position seems to be calculated by where the mouse is on the screen. That approach is okay, but it causes a sudden jump every time you drag the mouse over a window and back to the desktop. On Windows, it seems to be calculated by counting values of x/y movement, so it stops counting when the mouse is over a window, and it starts again when it's back on the desktop with no sudden jump. I hope it's possible to implement a system like on Windows here in the future :)

**Comment by duzhaokun123** [comment]
xdotool can do it https://github.com/jordansissel/xdotool/blob/735e301665e7f9b8fe850588e88a3a0973695eec/xdo.c#L891
maybe i can have a try

**Comment by Almamu** [comment]
@duzhaokun123  wouldn't a simple interpolation between current mouse position and last calculated position suffice? (this is pseudocode tho)
```
interpolationSpeed = 5
mouse = GetMousePosition() // returns current position of the mouse
x = lerp (x, mouse.x, interpolationSpeed) // this holds the "virtual" x position of the cursor for the background
y = lerp (y, mouse.y, interpolationSpeed) // this holds the "virtual" y position of the cursor for the background
```
that way you can produce the same effect @lunastrans was talking about

**Comment by duzhaokun123** [comment]
so we need a smooth anime when mouse back to desktop. not handle mouse moving when it on window. is it?

in fact i never tried wallpaper engine on Windows, im not sure how it handle mouse moving.

(sorry for my poor English)

**Comment by duzhaokun123** [comment]
well, i made it ugly https://github.com/duzhaokun123/wallpaper-engine-kde-plugin/commit/39f759e6380ac149db0aa64073f0f3364353e6aa
two modes, use comments to switch

**Comment by lunastrans** [comment]
> not handle mouse moving when it on window. is it?
Yes! I tried your implementation and it seems to be much smoother when moving over a window and way less distracting. It moves smoothly instead of jumping suddenly, so I definitely prefer it for now. 
But (unless I commented the wrong thing) it's not EXACTLY what I had in mind since it still pans over to the mouse position, just smoothly. The parallax effect on Windows seems to stop completely when over a window, @Almamu 's implementation would work for that - it was what I had in mind but I was too bad at coding to implement it/explain it. Although, I wonder how it would work with interactive wallpapers that display stuff over the mouse position (I don't know if this plugin supports them - I just remembered that Windows has them) - maybe it keeps a separate virtual position when using parallax wallpapers and uses your actual mouse position for other effects. Xdotool would work nicely for those probably, but there's still the problem of it being X-server exclusive.
